packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -242,6 +243,7 @@ export class PendingStateManager implements IDisposable {
 		);
 		this.pendingMessages.shift();
 
+		//* Needs to include compatDetails - Write failing test first
 		const messageContent = JSON.stringify({ type: message.type, contents: message.contents });
```

Comment:
@kian-thompson / @wes-carlson - When I added in the new `compatDetails` property, JSON.stringify serialized the props in a different order here compared to `pendingMessage.content`.  I got kind of nervous about doing that comparison here, although I guess it's working?

Looks like it was added in response to [this conversation](https://github.com/microsoft/FluidFramework/pull/13994/files#r1100677930), how do we know the order of props will be preserved on server roundtrip?  Is it reliable enough that I can just be sure to serialize the same way here to get it to pass the tests, and it'll be find in prod?

I put a Draft PR out for a helper to do a proper deep comparison of the objects (#16834), but it's probably slower than this check - although, I also wonder how slow this check is already, this is on the critical path, right?